src/widgets/scale/lv_scale_private.h

```diff
@@ -28,40 +28,49 @@ extern "C" {
  **********************/
 
 struct _lv_scale_section_t {
-    lv_style_t * main_style;
-    lv_style_t * indicator_style;
-    lv_style_t * items_style;
-    int32_t minor_range;
-    int32_t major_range;
-    uint32_t first_tick_idx_in_section;
-    uint32_t last_tick_idx_in_section;
-    uint32_t first_tick_idx_is_major;
-    uint32_t last_tick_idx_is_major;
-    int32_t first_tick_in_section_width;
-    int32_t last_tick_in_section_width;
-    lv_point_t first_tick_in_section;
-    lv_point_t last_tick_in_section;
+    lv_style_t * main_style;               /**< Style to use for MAIN part(s) of scale
+                                            *   when it falls within this section's range */
+    lv_style_t * indicator_style;          /**< Style to use for INDICATOR part(s) of scale
+                                            *   when it falls within this section's range */
+    lv_style_t * items_style;              /**< Style to use for ITEMS part(s) of scale
+                                            *   when it falls within this section's range */
+    int32_t range_min;                     /**< Scale parts with value >= this value will be drawn using applicable style. */
+    int32_t range_max;                     /**< Scale parts with value <= this value will be drawn using applicable style. */
+    uint32_t first_tick_idx_in_section;    /**< Internal (set during drawing): Tick index of first tick that falls within
+                                            *   this section; LV_SCALE_TICK_IDX_DEFAULT_ID if section contains no ticks. */
+    uint32_t last_tick_idx_in_section;     /**< Internal (set during drawing): Tick index of last tick that falls within
+                                            *   this section; LV_SCALE_TICK_IDX_DEFAULT_ID if section contains no ticks. */
+    int32_t first_tick_in_section_width;   /**< Internal (set during drawing) */
+    int32_t last_tick_in_section_width;    /**< Internal (set during drawing) */
+    lv_point_t first_tick_in_section;      /**< Internal (set during drawing) */
+    lv_point_t last_tick_in_section;       /**< Internal (set during drawing) */
+    uint32_t first_tick_idx_is_major : 1;  /**< Internal (set during drawing): true if
+                                            * `first_tick_idx_in_section` represents a major tick. */
```

Comment:
> The struct may reserve the whole uint32_t even if using 1 bit, is there any reason to limit the bit width? I'm afraid this is not portable across compilers

This is a good question.  I'd like to bring Gábor into this conversation.  I am mimicking what has been done in other similar structures.  @kisvegabor Thoughts?

Reply:
@uLipe My first thoughts about compiler compatibility is that this `struct` is used only internally, so it is okay if padding between fields changes from compiler to compiler since what uses it will have been compiled by the same compiler.  But I would still like to get Gábor's insights.

Reply:
We use bit fields on a lot of places and there was no complain about compiler issues so far.  

In this case there is no difference between 1 bit and `uint32_t` as the whole `uint32_t` will be used anyways. But if we added an other bit later, there wouldn't be extra memory usage.